diff --git a/sdl2-sys/build.rs b/sdl2-sys/build.rs
index fc4bd926..47e69483 100644
--- a/sdl2-sys/build.rs
+++ b/sdl2-sys/build.rs
@@ -223,7 +223,7 @@ fn link_sdl2(target_os: &str) {
             if cfg!(feature = "use_mac_framework") && target_os == "darwin" {
                 println!("cargo:rustc-flags=-l framework=SDL2");
             } else if target_os != "emscripten" {
-                println!("cargo:rustc-flags=-l SDL2");
+                println!("cargo:rustc-link-lib=SDL2");
             }
         }
     }
@@ -293,14 +293,14 @@ fn link_sdl2(target_os: &str) {
                 || target_os.contains("freebsd")
                 || target_os.contains("openbsd")
             {
-                println!("cargo:rustc-flags=-l SDL2_mixer");
+                println!("cargo:rustc-link-lib=SDL2_mixer");
             } else if target_os.contains("windows") {
-                println!("cargo:rustc-flags=-l SDL2_mixer");
+                println!("cargo:rustc-link-lib=SDL2_mixer");
             } else if target_os.contains("darwin") {
                 if cfg!(any(mac_framework, feature = "use_mac_framework")) {
                     println!("cargo:rustc-flags=-l framework=SDL2_mixer");
                 } else {
-                    println!("cargo:rustc-flags=-l SDL2_mixer");
+                    println!("cargo:rustc-link-lib=SDL2_mixer");
                 }
             }
         }
@@ -309,14 +309,14 @@ fn link_sdl2(target_os: &str) {
                 || target_os.contains("freebsd")
                 || target_os.contains("openbsd")
             {
-                println!("cargo:rustc-flags=-l SDL2_image");
+                println!("cargo:rustc-link-lib=SDL2_image");
             } else if target_os.contains("windows") {
-                println!("cargo:rustc-flags=-l SDL2_image");
+                println!("cargo:rustc-link-lib=SDL2_image");
             } else if target_os.contains("darwin") {
                 if cfg!(any(mac_framework, feature = "use_mac_framework")) {
                     println!("cargo:rustc-flags=-l framework=SDL2_image");
                 } else {
-                    println!("cargo:rustc-flags=-l SDL2_image");
+                    println!("cargo:rustc-link-lib=SDL2_image");
                 }
             }
         }
@@ -325,14 +325,14 @@ fn link_sdl2(target_os: &str) {
                 || target_os.contains("freebsd")
                 || target_os.contains("openbsd")
             {
-                println!("cargo:rustc-flags=-l SDL2_ttf");
+                println!("cargo:rustc-link-lib=SDL2_ttf");
             } else if target_os.contains("windows") {
-                println!("cargo:rustc-flags=-l SDL2_ttf");
+                println!("cargo:rustc-link-lib=SDL2_ttf");
             } else if target_os.contains("darwin") {
                 if cfg!(any(mac_framework, feature = "use_mac_framework")) {
                     println!("cargo:rustc-flags=-l framework=SDL2_ttf");
                 } else {
-                    println!("cargo:rustc-flags=-l SDL2_ttf");
+                    println!("cargo:rustc-link-lib=SDL2_ttf");
                 }
             }
         }
@@ -341,14 +341,14 @@ fn link_sdl2(target_os: &str) {
                 || target_os.contains("freebsd")
                 || target_os.contains("openbsd")
             {
-                println!("cargo:rustc-flags=-l SDL2_gfx");
+                println!("cargo:rustc-link-lib=SDL2_gfx");
             } else if target_os.contains("windows") {
-                println!("cargo:rustc-flags=-l SDL2_gfx");
+                println!("cargo:rustc-link-lib=SDL2_gfx");
             } else if target_os.contains("darwin") {
                 if cfg!(any(mac_framework, feature = "use_mac_framework")) {
                     println!("cargo:rustc-flags=-l framework=SDL2_gfx");
                 } else {
-                    println!("cargo:rustc-flags=-l SDL2_gfx");
+                    println!("cargo:rustc-link-lib=SDL2_gfx");
                 }
             }
         }
